codalab/lib/path_util.py

```diff
@@ -298,10 +299,33 @@ def rename(old_path, new_path):
     subprocess.call(['mv', old_path, new_path])
 
 
+def parse_azure_url(url):
```

Comment:
Let's get in the habit of type annotating everything.  Should we make a dataclass instead of using a tuple?  Eventually, I think we should migrate all our tuples over to dataclasses, which will make the code easier to understand.